### DISCUSSION TITLE
Fail to connect on 2023.10.1

### DISCUSSION THREAD

**Discussion by jdmevo123** [WillCodeForCats/solaredge-modbus-multi#452] [General]
Im getting complete failure to connect on 2023.10.1.

Logger: pymodbus.logging
Source: custom_components/solaredge_modbus_multi/hub.py:454
Integration: SolarEdge Modbus Multi ([documentation](https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki), [issues](https://github.com/WillCodeForCats/solaredge-modbus-multi/issues))
First occurred: 10:18:15 AM (221 occurrences)
Last logged: 3:10:26 PM

Failed to connect [Errno 111] Connect call failed ('192.168.86.28', 1502)
Failed to connect

**Comment by jdmevo123** [comment]
@WillCodeForCats  ive confirmed im connected via LAN and the IP is correct

**Comment by ChrisWarwick** [comment]
I have the same issue. I think this broke in 2023.09 but also fails in 2023.10 
I've reverted to 2023.08 for the time being.

**Comment by WillCodeForCats** [comment]
I have run all versions of 2023.09 and 2023.10 and have not observed or experienced any connection related issues.

**Comment by WillCodeForCats** [comment]
You can try [Release v2.4.7-pre.1](https://github.com/WillCodeForCats/solaredge-modbus-multi/releases/tag/v2.4.7-pre.1) which depends on the latest version of pymodbus which fixes a bug discovered in pymodbus older than 3.5.3 for a setting I use in the integration.

**Reply by ChrisWarwick** [reply]
Thanks - I've downloaded the pre-release (Wow! There's a lot of code there...)

I'm not sure if this is applicable to my install - also I'm a little unsure about how to install/test this. I'm away for a few days now but will look into this further next week. Thanks for getting back to me.

**Reply by WillCodeForCats** [reply]
You can install pre releases with HACS too, just turn on the show beta option.

**Reply by ChrisWarwick** [reply]
Thanks! 

While looking at this I noticed: (a) a new release of HA core, (b) some new releases of the integration
I've updated all the things and the error has gone. Not very helpful for you as it won't help future diagnosis, but I'm happy :-)

Thanks again

**Reply by ChrisWarwick** [reply]
Incidentally, fyi, I have a standard (domestic) single-phase inverter with a single SolarEdge battery. At some point I intend to try to be a little more controlling in terms of when/how much I charge from the grid based on sun forecasts etc (inspired by chats on Mastodon...)

I'm also about to set up a Home Assistant instance for a commercial operation I'm involved with - two 3-phase inverters and 2 batteries.

I'm very willing to feed back any config information or anything else you might find helpful at any point.

Cheers,
Chris
